com.unity.render-pipelines.core/Runtime/Utilities/ArrayUtils.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using Unity.Collections;
+using UnityEngine.Jobs;
+
+namespace UnityEngine.Rendering
+{
+    /// <summary>
+    /// Array utilities functions
+    /// </summary>
+    public static class ArrayUtils
```

Comment:
This seems useful. You could consider making them extension methods, so you could just say 
`decalToWorlds.Resize(newCapacity);` instead of  `ArrayUtils.ResizeNativeArray(ref decalToWorlds, newCapacity);`. Though on second thought I'm not certain if it's good practice to have such things in this namespace or if they would belong in a separate extension namespace instead.

Reply:
Hey so in c# you cant add methods / functions directly in a namespace. The way to do it is to have a static class, which is what I've done.
Let me know if this makes sense.

Reply:
Ah ok I had no idea this existed in c# :) 
The question is can it work with refs on structs (notice these functions replace the refs of the arrays)? If not then we cant use it. Ill try it now and see if it works.

Reply:
@arttu-peltonen I was successful with refs of structs, but not refs of managed arrays. Thus, this is still gonna look like a static method syntax for managed arrays (since I have to instantiate inside this method lazilly). Otherwise native arrays and tansform array access are now with the syntax you suggested. Tested URP to ensure it didnt break.

Reply:
Ok good to know there's a limitation like that. Thanks for considering it!